### ISSUE TITLE
Projects SSH Key stops working after a reboot

### ISSUE THREAD

**Issue by binaryn3xus** [node-red/node-red#4488]
### Current Behavior

Currently if you go through setting up your SSH key and then you need to reboot the container for any reason, the SSH key stops working. It will give you a popup asking for the passphrase (i didnt set one).

### Expected Behavior

I would expect the SSH key to continue to work after a simple restart of the container.

### Steps To Reproduce

1) Assuming you already have your Project setup
2) Add SSH Key
3) Test key by doing a push to the repo (Assume successful here)
4) Restart container
5) Try to make a small change, deploy, and then go to git history and try to push it.
6) Get the popup for SSH passphrase (clicking retry does not work).

### Example flow

N/A

### Environment

- Node-RED version: v3.1.3
- Node.js version: v16.20.2
- npm version:
- Platform/OS: Linux (Container)
- Browser: Firefox

- SSH URL: ssh://git@github.com/binaryn3xus/My-Private-Repo-Name.git

**Comment by hardillb** [comment]
Please supply full details of how you started the container and what volumes you have mounted where.

Also supply exact details of how you restarted the container.

We will also move this to the node-red-docker repo

**Comment by binaryn3xus** [comment]
Sorry. My apologizes on posting this on the wrong repo for Node-Red. I hope you can move it. If not, I can create a new issue in the right place.

As for the details on the container, I am running K3s so its in a Kubernetes cluster. You can see my configurations and such here: [https://github.com/binaryn3xus/HomeOps/blob/main/kubernetes/apps/default/node-red/app/helmrelease.yaml](https://github.com/binaryn3xus/HomeOps/blob/main/kubernetes/apps/default/node-red/app/helmrelease.yaml)

I am requesting the container to shutdown then restarting it.

**Comment by hardillb** [comment]
Just for clarity here.

- You generated new ssh keys using the wizard under settings -> git config?
- If you create a shell in the container can you look in `/data/projects/.sshkesy` do you see 2 files (one ending in `.pub`
- Also is there anything in the container logs from the time you are prompted for the password?

**Comment by binaryn3xus** [comment]
Yes, i am generating from the wizard in the UI

```shell
node-red-7957f647b7-dmr7w:/data/projects$ ls -a /data/projects/.sshkeys
.                                   ..                                  __default_NodeRedHomeOpsSSHKey      __default_NodeRedHomeOpsSSHKey.pub
node-red-7957f647b7-dmr7w:/data/projects$
```

Logs:

Okay so today it is weird...It fails the first time and then the next it works. I promise though, it was failing repetitively...

```logs

18 Dec 12:29:03 - [info] Started flows                                                                                                                                  
18 Dec 12:29:03 - [trace] runtime event: {"id":"runtime-deploy","payload":{"revision":"ee62b44bc489da39d619fd59d4e225db"},"retain":true}                                
18 Dec 12:29:03 - [info] [server:Home Assistant (User: NodeRed)] Connecting to http://home-assistant.default.svc.cluster.local:8123                                     
18 Dec 12:29:03 - [trace] git["rev-list","HEAD","--count"]                                                                                                              
18 Dec 12:29:03 - [trace] git["rev-list","HEAD","--count"]                                                                                                              
18 Dec 12:29:03 - [trace] git["log","--format=sha: %H%nparents: %p%nrefs: %D%nauthor: %an%ndate: %ct%nsubject: %s%n-----","-n 20"]                                      
18 Dec 12:29:03 - [info] [mqtt-broker:Kubernetes MQTT Broker] Connected to broker: Node-Red@mqtt://mqtt.REDACTED.com:1883                                              
18 Dec 12:29:03 - [trace] git["ls-files","--cached","--others","--exclude-standard"]                                                                                    
18 Dec 12:29:03 - [trace] git["status","--porcelain","-b"]                                                                                                              
18 Dec 12:29:03 - [info] [server:Home Assistant (User: NodeRed)] Connected to http://home-assistant.default.svc.cluster.local:8123                                      
18 Dec 12:29:04 - [debug] [server:Home Assistant (User: NodeRed)] States Loaded                                                                                         
18 Dec 12:29:04 - [debug] [server:Home Assistant (User: NodeRed)] Services Loaded                                                                                       
18 Dec 12:29:04 - [debug] [server:Home Assistant (User: NodeRed)] HA State: running                                                                                     
18 Dec 12:29:04 - [debug] [server:Home Assistant (User: NodeRed)] Integration: notloaded                                                                                
18 Dec 12:29:05 - [debug] [api-call-service:Turn off FP Light] Calling Service: {"domain":"light","service":"turn_off","target":{"entity_id":"light.front_porch_light_le
18 Dec 12:29:05 - [debug] [api-call-service:Turn Off Christmas Tree] Calling Service: {"domain":"switch","service":"turn_off","target":{"entity_id":"switch.ext_lr_outle
18 Dec 12:29:05 - [debug] [api-call-service:Turn Off Porch Outlet] Calling Service: {"domain":"switch","service":"turn_off","target":{"entity_id":"switch.front_porch_ou
18 Dec 12:29:10 - [debug] [api-call-service:Turn off Holiday Mains] Calling Service: {"domain":"switch","service":"turn_off","target":{"entity_id":"switch.holidaylights
18 Dec 12:29:14 - [trace] git["push","origin","HEAD:main","--porcelain"]                                                                                                
18 Dec 12:29:14 - [debug] git -c credential.helper= push origin HEAD:main --porcelain                                                                                   
18 Dec 12:29:14 - [debug] [err] git@github.com: Permission denied (publickey).                                                                                          
18 Dec 12:29:14 - [debug] [err] fatal: Could not read from remote repository.                                                                                           
18 Dec 12:29:14 - [debug] [err]                                                                                                                                         
18 Dec 12:29:14 - [debug] [err] Please make sure you have the correct access rights                                                                                     
18 Dec 12:29:14 - [debug] [err] and the repository exists.                                                                                                              
18 Dec 12:29:14 - [debug] rc=128                                                                                                                                        
18 Dec 12:29:17 - [trace] runGitCommandWithSSHCommand {"keyFile":"NodeRedHomeOpsSSHKey","passphrase":"[hidden]"key_path":"/data/projects/.sshkeys/__default_NodeRedHomeO
18 Dec 12:29:17 - [trace] git["push","origin","HEAD:main","--porcelain"]                                                                                                
18 Dec 12:29:17 - [debug] git -c credential.helper= push origin HEAD:main --porcelain                                                                                   
18 Dec 12:29:18 - [debug] [out] To ssh://github.com/binaryn3xus/HomeOps-Node-Red-Automation.git                                                                         
18 Dec 12:29:18 - [debug] [out]      HEAD:refs/heads/main    35fdfdf..faff52b                                                                                           
18 Dec 12:29:18 - [debug] [out] Done                                                                                                                                    
18 Dec 12:29:18 - [debug] rc=0                                                                                                                                          
18 Dec 12:29:18 - [trace] git["rev-list","HEAD","--count"]                                                                                                              
18 Dec 12:29:18 - [trace] git["rev-list","HEAD","--count"]                                                                                                              
18 Dec 12:29:18 - [trace] git["log","--format=sha: %H%nparents: %p%nrefs: %D%nauthor: %an%ndate: %ct%nsubject: %s%n-----","-n 20"]                                      
18 Dec 12:29:18 - [trace] git["ls-files","--cached","--others","--exclude-standard"]                                                                                    
18 Dec 12:29:19 - [trace] git["status","--porcelain","-b"]
```

**Comment by hardillb** [comment]
OK, so iirc the password is never stored on disk, it's only ever kept in RAM so a restart will clear that. So the first time after a restart it should ask for the password (and you should be able to just submit an empty password if that is correct).

**Comment by binaryn3xus** [comment]
And that would be fine. But before I regenerated a new SSH key, it kept failing even when retrying the attempt.

The previous ssh key was still valid and not expired. I don't know if it was just some strange permission issue or something else. I didnt have a lower logging level before now. Im going to leave it on `trace` unless you recommend just `debug` level and keep any eye on it.

**Comment by hardillb** [comment]
Ahh, it wasn't clear you'd generated an other SSH key here. 

Looking at the code, there is nothing logged at anything more detailed than `trace` so that should be fine.

**Comment by hardillb** [comment]
As a suggestion, can you copy the private key out of the container and just double check you can use it to connect to the git server without needing to supply a password.

`ssh -i /path/to/key git@gitserver`

Actually you can probably run that in the container as well rather than copying it out (and the hostnames will probably work if they are kubernetes internal DNS names.

**Comment by binaryn3xus** [comment]
Like this?

```log
node-red-68d8456db-kckwq:~$ ssh -i /data/projects/.sshkeys/__default_NodeRedHomeOpsSSHKey git@github.com
PTY allocation request failed on channel 0
Hi binaryn3xus! You've successfully authenticated, but GitHub does not provide shell access.
Connection to github.com closed.
```

**Comment by hardillb** [comment]
Yes, but I was hoping for access to the original failing keys, if you have overwritten them when you created new keys then it's not going to show if there was a problem with them.

**Comment by binaryn3xus** [comment]
I guess this issue is going to make a liar out of me. haha.

I have tried several times again to reproduce it. I guess it must have been caused by something else. I have had this happen several times in the past though. So maybe I will just leave the logging on `trace` and see if it comes back or not.

I will close this issue for now and ask to be reopened if the problem returns. Thanks for the help @hardillb
